[WebDriver] Update imported WPT WebDriver tests

lauromoura opened the pull request:
#### 1f76c86e781a3ceff89d5bcb081e2a148e53ef1e
<pre>
[WebDriver] Update imported WPT WebDriver tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=267835">https://bugs.webkit.org/show_bug.cgi?id=267835</a>

Reviewed by Carlos Garcia Campos.

Bump imported version to 097898330ea0 from late January.

Upstream <a href="https://github.com/web-platform-tests/wpt/pull/40910">https://github.com/web-platform-tests/wpt/pull/40910</a> changed
WPT WebDriver tests to use HTTPS urls instead of HTTP. This led to tests
using URLs provided by the local server to fail with unrecognized
certificates.

This commit forces the browser under test to use the test certificates
instead of relying on the system ones. While this allows using the local
HTTPS server, it has the drawback of ingnoring the system certificates,
failing to access external websites if a test wants to (which does not
seems to be the case now).

Also rename the config `bind_hostname` to `bind_address` after
<a href="https://github.com/web-platform-tests/wpt/pull/9998">https://github.com/web-platform-tests/wpt/pull/9998</a>

* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_executor.py:
(WebDriverW3CExecutor.__init__): Export variable for certificate setup
(WebDriverW3CExecutor._runner): Add new timeout_multipler key as in WTP 1a44d855cb
* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_web_server.py:
(WebDriverW3CWebServer.start): Store information about certificates
being used in the current test run.
(WebDriverW3CWebServer.cacert_pem_file): Export the certificates
location
* WebDriverTests/TestExpectations.json: Update classic and new bidi
  expectations
* WebDriverTests/imported/w3c/config.json: Rename bind_hostname to
  bind_address key
* WebDriverTests/imported/w3c/importer.json: Update imported WPT hash
* WebDriverTests/imported/w3c/tools/: Bump imported code
* WebDriverTests/imported/w3c/webdriver/: Bump imported code

Canonical link: <a href="https://commits.webkit.org/276326@main">https://commits.webkit.org/276326@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/785ce860ffa4d620cc03f160809c07c8c7f498de

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44107 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23176 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46540 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46749 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40134 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46412 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27102 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20569 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36346 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44685 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20201 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37980 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17374 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/43984 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17682 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39081 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2157 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40271 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39369 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48337 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19085 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15658 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43204 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20446 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41927 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9862 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20671 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20072 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos reviewed (approved):
Amazing work here. Thanks!

lauromoura committed:
[WebDriver] Update imported WPT WebDriver tests
https://bugs.webkit.org/show_bug.cgi?id=267835

Reviewed by Carlos Garcia Campos.

Bump imported version to 097898330ea0 from late January.

Upstream https://github.com/web-platform-tests/wpt/pull/40910 changed
WPT WebDriver tests to use HTTPS urls instead of HTTP. This led to tests
using URLs provided by the local server to fail with unrecognized
certificates.

This commit forces the browser under test to use the test certificates
instead of relying on the system ones. While this allows using the local
HTTPS server, it has the drawback of ingnoring the system certificates,
failing to access external websites if a test wants to (which does not
seems to be the case now).

Also rename the config `bind_hostname` to `bind_address` after
https://github.com/web-platform-tests/wpt/pull/9998

* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_executor.py:
(WebDriverW3CExecutor.__init__): Export variable for certificate setup
(WebDriverW3CExecutor._runner): Add new timeout_multipler key as in WTP 1a44d855cb
* Tools/Scripts/webkitpy/webdriver_tests/webdriver_w3c_web_server.py:
(WebDriverW3CWebServer.start): Store information about certificates
being used in the current test run.
(WebDriverW3CWebServer.cacert_pem_file): Export the certificates
location
* WebDriverTests/TestExpectations.json: Update classic and new bidi
  expectations
* WebDriverTests/imported/w3c/config.json: Rename bind_hostname to
  bind_address key
* WebDriverTests/imported/w3c/importer.json: Update imported WPT hash
* WebDriverTests/imported/w3c/tools/: Bump imported code
* WebDriverTests/imported/w3c/webdriver/: Bump imported code

Canonical link: https://commits.webkit.org/276326@main
